lib/extensions/ar_virtual.rb

```diff
@@ -121,7 +121,7 @@ def virtual_delegate_arel(col, to, to_model, to_ref)
           end
         elsif to_ref.macro == :belongs_to
           lambda do |t|
-            src_model_id = arel_attribute(to_ref.association_foreign_key, t)
+            src_model_id = arel_attribute(to_ref.foreign_key, t)
```

Comment:
Should this be fixed in a separate PR?  I realize it is related, but it is a much larger thing we are changing here then focusing specifically ownership.